mitiq/interface/mitiq_qibo/conversions.py

```diff
@@ -0,0 +1,394 @@
+
+
+GATES_TO_DECOMPOSE = {
+    "crx": crx_decomp,
+    "cry": cry_decomp,
+    "crz": crz_decomp,
+    "cu1": cu1_decomp,
+    "cu3": cu3_decomp,
+    "csx": csx_decomp,
+    "csxdg": csxdg_decomp,
+    "iswap": iswap_decomp,
+    "fswap": fswap_decomp,
+    "rxx": rxx_decomp,
+    "ryy": ryy_decomp,
+    "rzz": rzz_decomp,
+}
+
+
+class UnsupportedQiboCircuitError(Exception):
+    pass
+
+
+def decompose_qibo_circuit(qibo_circuit: QiboCircuit) -> QiboCircuit:
+    """Returns a QiboCircuit circuit equivalent to the input QiboCircuit
+    with all unknown by cirq gates decomposed.
+
+    Args:
+        qibo_circuit: QiboCircuit to decompose.
+
+    Returns:
+        Decomposed QiboCircuit
+    """
+    decomp_circuit = QiboCircuit(qibo_circuit.nqubits)
+    for gate in qibo_circuit.queue:
+        decomposition_func = GATES_TO_DECOMPOSE.get(gate.name, lambda x: x)
+        decomp_circuit.add(decomposition_func(gate))
+
+    return decomp_circuit
+
+
+def from_qibo(qibo_circuit: QiboCircuit) -> Circuit:
+    """Returns a Cirq circuit equivalent to the input QiboCircuit.
+
+    Args:
+        qibo_circuit: QiboCircuit to convert to a Cirq circuit.
+
+    Returns:
+        Cirq circuit representation equivalent to the input QiboCircuit.
+    """
+    for measurement in qibo_circuit.measurements:
+        reg_name = measurement.register_name
+        if not reg_name.islower():
+            raise UnsupportedQiboCircuitError(
+                f"OpenQASM does not support capital letters in "
+                f"register names but {reg_name} was used."
+            )
+    for gate in qibo_circuit.queue:
+        if isinstance(gate, gates.M):
+            continue
+        if gate.is_controlled_by:
+            raise UnsupportedQiboCircuitError(
+                "OpenQASM does not support multi-controlled gates."
+            )
+        try:
+            gate.qasm_label
+        except NotImplementedError:
+            raise UnsupportedQiboCircuitError(
+                f"{gate.name} is not supported by OpenQASM."
+            )
+
+    # Decompose the circuit in known Cirq gates
+    qasm = decompose_qibo_circuit(qibo_circuit).to_qasm()
+
+    return cirq_from_qasm(qasm)
+
+
+def to_qibo(circuit: Circuit) -> QiboCircuit:
+    """Returns a QiboCircuit equivalent to the input Cirq circuit.
+
+    Args:
+        circuit: Cirq circuit to convert to a QiboCircuit.
+
+    Returns:
+        QiboCircuit object equivalent to the input Mitiq circuit.
+    """
+    qasm = cirq_to_qasm(circuit)
+    # Remove problematic comment lines in the qasm code
+    qasm = "\n".join(
+        [
+            line
+            for line in qasm.split("\n")
+            if not line.strip().startswith("//")
+        ]
+    )
+    # Remove problematic spaces in gate arguments
+    qasm = "\n".join([line.replace(", ", ",") for line in qasm.split("\n")])
+    # Remove in line problematic comments
+    lines = qasm.split("\n")
+    clean_lines = []
+    for line in lines:
```

Comment:
I think we can avoid converting between strings and lists a few times here by keeping things as a list after the first conversion. WDYT? (code not tested and formatting may be off!)

```suggestion
    # Remove problematic comment lines in the qasm code
    qasm = [
        line.replace(", ", ",") # remove spaces in gate arguments
        for line in qasm.split("\n")
        if not line.strip().startswith("//")
    ]
    
    clean_lines = []
    for line in qasm:
```

Reply:
The thing with multi-controlled gates is that in the` QiboCircuit.to_qasm()` function from Qibo, if the gate is controlled by a similar error is raised. They do not comment much on why they raise the error though.
https://github.com/qiboteam/qibo/blob/fb55eee2eca731a10fa97a6d2195ee1fa65f5b9a/src/qibo/models/circuit.py#L1160

Reply:
I've managed to avoid some of the unnecessary back-and-forth conversions between lists and strings, but I had to edit the suggested code a bit since mypy was complaining about the inconsistent type of `qasm`.